astroid/decorators.py

```diff
@@ -70,6 +70,13 @@ class cachedproperty:
     __slots__ = ("wrapped",)
 
     def __init__(self, wrapped):
+        if sys.version_info >= (3, 8):
+            warnings.warn(
+                "The cachedproperty class is functionally the same as functools.cached_property "
+                "and will be removed after support for Python < 3.8 has been dropped. "
+                "Consider importing the functools variant on >= 3.8.",
```

Comment:
I would actually suggest to remove it with astroid `3.0` (at least for Python 3.8+). We can define it conditionally after that point.
```py
if sys.version_info < (3, 8):
    class cachedproperty:
        ...
```

My suggestion
```
cachedproperty has been deprecated and will be removed in astroid 3.0 for Python 3.8+.
Use functools.cached_property instead.
```

--
As a side note, personally I do consider `cachedproperty` to be internal. Thus the deprecation note is more for us than anyone else. It might even make sense to add a `TODO` along the lines: `Remove completely when support for 3.7 is dropped`.

Reply:
I didn't know if `3.0` would be `3.8+` so I didn't mention a specific version. I'll update.